packages/js/src/utils/typescript/ts-config.ts

```diff
@@ -85,6 +88,17 @@ export function updateRootTsConfig(
       ),
     ];
 
+    if (options.additionalImportPaths) {
+      for (const additionalImportPath of options.additionalImportPaths) {
+        c.paths[`${options.importPath}/${additionalImportPath}`] = [
+          joinPathFragments(
+            options.projectRoot,
+            `./src/${additionalImportPath}.${options.js ? 'js' : 'ts'}`
+          ),
+        ];
+      }
+    }
+
```

Comment:
I don't know if I'm a huge fan of having this here..

Not a big deal probably... but this currently doesn't check if those paths already exist and would just overwrite them :shrug:. 

I feel like we should rework this function's API to be more like `addTsConfigImportPaths(tree, importPath: string, importModules: string[])`

```ts
function jsLibraryGenerator(tree) {
  addTsConfigPath(tree, importPath, [joinPathFragments(projectRoot, 'src/index.ts')])
  addTsConfigPath(tree, joinPathFragments(importPath, 'server'), [joinPathFragments(projectRoot, 'src/server.ts')])
}
```

This would open us up to adding import paths which are not under `src`.

We could add the new function and keep the current usages the same.. then transition them over. What do you think?

Reply:
Yeah we can do something like that.

Reply:
Removed the original function and added `addTsConfigPath` so each library generator can add the import paths that are needed. For now, the new `server.ts` entry is only for Next.js apps, and we can look at adding something similar to `@nrwl/react:lib` later when it's more clear the direction of RSCs.